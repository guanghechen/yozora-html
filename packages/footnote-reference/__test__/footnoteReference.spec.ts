import type { FootnoteReference as IFootnoteReference } from '@yozora/ast'
import renderFootnoteReference from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node = {
      type: 'footnoteReference',
      identifier: 'bravo',
      label: 'bravo',
    }
    expect(renderFootnoteReference(node as IFootnoteReference)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node = {
      type: 'footnoteReference',
      identifier: 'bravo"',
      label: 'bravo<a>"<b:"></sup>',
    }
    expect(renderFootnoteReference(node as IFootnoteReference)).toMatchSnapshot()
  })
})
