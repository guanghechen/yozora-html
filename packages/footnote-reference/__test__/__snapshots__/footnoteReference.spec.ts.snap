// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<sup id=\\"reference-bravo\\" class=\\"yozora-footnote-reference\\"><a href=\\"#bravo\\" title=\\"bravo\\">[bravo]</a></sup>"`;

exports[`snapshot basic-2 1`] = `"<sup id=\\"reference-bravo%22\\" class=\\"yozora-footnote-reference\\"><a href=\\"#bravo%22\\" title=\\"bravo\\"\\">[bravo\\"]</a></sup>"`;
